### DISCUSSION TITLE
Not working as expected 🤔

### DISCUSSION THREAD

**Discussion by MestreKarin** [Automattic/mongoose#14814] [Q&A] [answered]
I may be missing something because some of my code is not working as expected. For example:

```typescript
interface IAccount {
  name: String;
  email: String;
  password: String;
}

const AccountSchema = new Schema<IAccount>();
export const Account = model('account', AccountSchema);

// -- IN ANOTHER FILE --

const createdAccount = await Account.create({
  name: 'Hello World',
  email: 'foo@bar.com',
  password: 'strong1234'
});

console.log(createdAccount);

// -- OUTPUT --
{ _id: new ObjectId('66c17b5d4a5edf95928b918d'), __v: 0 }
```

According to the [docs](https://mongoosejs.com/docs/api/model.html#Model.create()) this should work, however, none of the properties are saved (I checked with Compass).

I got a strange behavior when selecting too:

```typescript
const existingAccount = await Account.findOne({ email: 'foo@bar.com' });

console.log(existingAccount);
/*
{
  _id: new ObjectId('66be3f54fff3e68e10d4b06c'),
  name: 'Hello World',
  email: 'foo@bar.com',
  password: 'strong1234'
}
*/

console.log(existingAccount.name); // --> undefined
console.log(existingAccount.toObject().name); // --> 'Hello World'

const existingAccountTwo = await Account.findOne({ email: 'foo@bar.com' }).lean();

console.log(existingAccountTwo.name); // --> 'Hello World'
```

Although the [docs](https://mongoosejs.com/docs/api/model.html#Model.findOne()) use `.exec()`, it makes no difference, I either need to use `.lean()` or `.toObject()` to read the document properties.

My node version is `v22.0.0` and the mongoose is `8.5.3`. I need some light 😔

**Comment by MestreKarin** [comment, marked as answer]
I ended up using the mongodb driver.

**Comment by vkarpov15** [comment]
The issue is that your schema is empty: `const AccountSchema = new Schema<IAccount>();`, you need to define your schema so Mongoose knows which getters to define on your documents
